src/main/java/org/candlepin/subscriptions/event/EventConflictResolver.java

```diff
@@ -0,0 +1,221 @@
+    Map<String, Double> deductions = new HashMap<>();
+    conflictingEvents.stream()
+        .map(EventRecord::getEvent)
+        .forEach(
+            e ->
+                e.getMeasurements()
+                    .forEach(
+                        m -> {
+                          deductions.putIfAbsent(m.getUom(), 0.0);
+                          deductions.put(m.getUom(), deductions.get(m.getUom()) - m.getValue());
+                        }));
+    return deductions;
+  }
+
+  private List<EventRecord> resolveEventMeasurements(
+      Event toResolve, Map<String, Double> measurementDeductions) {
+    List<EventRecord> resolved = new ArrayList<>();
+
+    // If there is nothing to deduct from the event to resolve,
+    // simply return the event as is.
+    if (measurementDeductions.isEmpty()) {
+      log.debug("Nothing to resolve for event {}", toResolve);
+      resolved.add(new EventRecord(toResolve));
+      return resolved;
+    }
+
+    // Measurements need to be resolved individually since conflicting events
+    // could potentially contain different uom measurement sets.
+    Set<String> measurementUomsToSkip = new HashSet<>();
+    for (Measurement measurement : toResolve.getMeasurements()) {
+      Optional<EventRecord> deductionEvent =
+          buildDeductionEvent(
+              toResolve,
+              measurement,
+              Optional.ofNullable(measurementDeductions.get(measurement.getUom())));
+      deductionEvent.ifPresentOrElse(
+          resolved::add, () -> measurementUomsToSkip.add(measurement.getUom()));
+    }
+
+    rebuildEventWithUpdatedMeasurements(toResolve, measurementUomsToSkip).ifPresent(resolved::add);
+    return resolved;
+  }
+
+  private Map<EventKey, List<EventRecord>> getConflictingEvents(List<Event> incomingEvents) {
+    List<EventKey> eventKeys = incomingEvents.stream().map(EventKey::fromEvent).toList();
+    return eventRecordRepository.findConflictingEvents(eventKeys).stream()
+        .collect(Collectors.groupingBy(e -> EventKey.fromEvent(e.getEvent())));
+  }
+
+  private Optional<EventRecord> buildDeductionEvent(
+      Event toResolve, Measurement measurement, Optional<Double> deduction) {
+    if (deduction.isPresent()) {
+      Double deductedValue = deduction.get();
+      // If we had a conflicting Event, but the measurement value was the same
+      // there's nothing to resolve as they are considered equal. This will
+      // NOT result in a new EventRecord for the measurement.
+      //
+      // The deducted value is expected to be <= 0, so we compare
+      // by adding the incoming value to it to see if the result is 0.
+      if (deductedValue + measurement.getValue() == 0.0) {
+        log.debug("Incoming event measurement is a duplicate. Nothing to resolve.");
+        return Optional.empty();
+      }
+
+      // Create an EventRecord with the -ve value
+      Event deductionEvent = createRecordFrom(toResolve);
```

Comment:
I would like to quickly identify the events that are deductions (so, we can easily identify them via queries in case we need to amend the logic in the future).
Perhaps, adding a new field in the Event pojo to state that this is a deducted-event or something like this.

Reply:
DONE - next push.